erpnext/selling/doctype/quoting_sheet/quoting_sheet.js

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+frappe.provide("erpnext.quoting_sheet")
+frappe.ui.form.on('Quoting Sheet', {
+	refresh: function(frm) {
+		console.log("hello refresh")
+	},
+	item_code: function(frm) {
+		if(frm.doc.currency == frappe.sys_defaults.currency){
+			frm.set_value("conversion_rate", 1.0)
+		}
+	},
+	invoke_doc_function(frm, method) {
+		frappe.call({
+			doc: frm.doc,
+			method: method,
+			callback: function(r) {
+				if(!r.exe) {
+					frm.refresh_fields();
+				}
+			}
+		});
+	},
+	calculate: function(frm) {
+		frm.events.invoke_doc_function(frm, "calculate_totals");
+	},
+});
+
+frappe.ui.form.on("BOM Item", {
+	item_code: function(frm,cdt,cdn) {
+		var row = locals[cdt][cdn];
+		frappe.call({
+			method: "erpnext.selling.doctype.quoting_sheet.quoting_sheet.get_item_details",
+			args: {
+				"item_code": row.item_code
+			},
+			callback: (res)=> {
+				console.log(res)
+				frappe.model.set_value(cdt,cdn,"rate",res.message[0].valuation_rate)
+				frappe.model.set_value(cdt,cdn,"bom_no",res.message[0].default_bom)
+				frappe.model.set_value(cdt,cdn,"uom",res.message[0].stock_uom)
+			},
+		})
+		console.log("ROW", row)
+																																		
+	},
+
+	qty: function(frm,cdt,cdn) {
+		var row = locals[cdt][cdn];
+		if(row.qty && row.rate) {
+			let amount = row.qty * row.rate
+			frappe.model.set_value(cdt,cdn,"amount", amount)
+		}
+	},
+	amount: function(frm,cdt,cdn) {
+		var rm_items = frm.doc.rm_items;
+		let cost = 0;
+		rm_items.forEach(element => {
+			cost += element.amount
+		});
+		console.log("amount", cost);
+		frm.set_value("rm_cost", cost);
+
+
+	}
+
+
+});
+
+frappe.ui.form.on("BOM Item", "qty", function(frm, cdt, cdn) {
+	var d = locals[cdt][cdn];
+	d.stock_qty = d.qty * d.conversion_factor;
+	refresh_field("stock_qty", d.name, d.parentfield);
+});
```

Comment:
@hrwX why are we removing this functionality? In case it has to do with the child table, we are using the same child doctype called "BOM Item" in the quoting sheet as it is exactly the same and we didn't have a need to create another child table